File: ScpProject/StrengthConApp/src/app/Models/Athlete/CompletedSet.ts
export class CompletedSet {
    public Id: number;
    public Position: number;
    public Sets: number;
    public Reps: number;
    public Percent: number;
    public Weight: number;
    public AssignedProgramId: number;
    public OriginalSetId: number
    public AthleteId : number;
}

export class CompletedSuperSet {
    public Id: number;
    public Position: number;
    public Sets: number;
    public Reps: number;
    public Percent: number;
    public Weight: number;
    public AssignedProgramId: number;
    public OriginalSuperSet_SetId: number
    public AthleteId : number;
    public RepsAchieved : number; 
}

File: ScpProject/Models/Athlete/CompletedSet.cs

using System.ComponentModel.DataAnnotations.Schema;
using System;

namespace Models.Athlete
{
    public class CompletedSet
    {
        public int Id { get; set; }
        public int Position { get; set; }
        public int Sets { get; set; }
        public int Reps { get; set; }
        public double Percent { get; set; }
        public double Weight { get; set; }
        public int OriginalSetId { get; set; }
        public int AthleteId { get; set; }
        public int AssignedProgramId { get; set; }
        public DateTime CompletedDate { get; set; }
        [ForeignKey("OriginalSetId")]
        public virtual Program.ProgramSet OriginalSet { get; set; }

        [ForeignKey("AthleteId")]
        public virtual Athlete TargetAthlete { get; set; }
        [ForeignKey("AssignedProgramId")]
        public Program.AssignedProgram AssignedProgram { get; set; }
    }
}
